From nobody Mon Sep 17 00:00:00 2001
From: Juergen Schindele <schindele@nentec.de>
Date: Mon Jan 9 08:51:48 2006 +0100
Subject: [PATCH] USB: touchkitusb.c (eGalax driver) fix
Patch-mainline: 2.6.16-rc3

This patch corrects the URB initialisation for transfers
like this is done in other drivers too.
Without this patch no data was transmitted on a PXA270 OHCI
platform. May apply to others too.

Signed-off-by: Juergen Schindele <schindele@nentec.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---

 drivers/usb/input/touchkitusb.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

a63d3cb8ba662c02c27c655f37de05f73c0db98c
diff --git a/drivers/usb/input/touchkitusb.c b/drivers/usb/input/touchkitusb.c
index 3b3c7b4..697c5e5 100644
--- a/drivers/usb/input/touchkitusb.c
+++ b/drivers/usb/input/touchkitusb.c
@@ -337,6 +337,9 @@ static int touchkit_probe(struct usb_int
 			 touchkit->data, TOUCHKIT_REPORT_DATA_SIZE,
 			 touchkit_irq, touchkit, endpoint->bInterval);
 
+	touchkit->irq->transfer_dma = touchkit->data_dma;
+	touchkit->irq->transfer_flags |= URB_NO_TRANSFER_DMA_MAP;
+
 	input_register_device(touchkit->input);
 
 	usb_set_intfdata(intf, touchkit);
-- 
1.1.4.gdcc6

